3rdparty/libgcov/src/GCDAProfiling.c

```diff
@@ -0,0 +1,645 @@
+/*===- GCDAProfiling.c - Support library for GCDA file emission -----------===*\
+|*
+|*                     The LLVM Compiler Infrastructure
+|*
+|* This file is distributed under the University of Illinois Open Source
+|* License. See LICENSE.TXT for details.
+|* 
+|*===----------------------------------------------------------------------===*|
+|* 
+|* This file implements the call back routines for the gcov profiling
+|* instrumentation pass. Link against this library when running code through
+|* the -insert-gcov-profiling LLVM pass.
+|*
+|* We emit files in a corrupt version of GCOV's "gcda" file format. These files
+|* are only close enough that LCOV will happily parse them. Anything that lcov
+|* ignores is missing.
+|*
+|* TODO: gcov is multi-process safe by having each exit open the existing file
+|* and append to it. We'd like to achieve that and be thread-safe too.
+|*
+\*===----------------------------------------------------------------------===*/
```

Comment:
Should you include the license at the top of this file?

Reply:
All the files (including this one) in the directory are directly copied from [LLVM compiler-rt](https://github.com/llvm-mirror/compiler-rt/tree/master/lib/profile). Not sure if I need to add license if they didn't come with it. In this PR, I put the LICENSE file of the compiler-rt under 3rdparty/libgcov/.